src/routes/auth/providers/utils.ts

```diff
@@ -113,14 +123,28 @@ export const initProvider = <T extends Strategy>(
 
   subRouter.get('/', passport.authenticate(strategyName, { session: false }))
 
-  subRouter.get(
-    '/callback',
-    passport.authenticate(strategyName, {
-      failureRedirect: PROVIDERS_FAILURE_REDIRECT,
-      session: false
-    }),
-    providerCallback
-  )
+  // The Sign in with Apple auth provider requires a POST route for authentication
+  if (strategyName === 'apple') {
+    subRouter.post(
+      '/callback',
+      express.urlencoded(),
```

Comment:
Do we need this for other (non apple) providers too? What's the purpose for the `urlencoded()` here?

Reply:
Yes, looks like it's needed without I don't get it working. I have made a opt-in for auth providers in the latest version